src/packer.ts

```diff
@@ -111,9 +158,17 @@ export const pack = <Type extends readonly string[]>(
       const parser = getParser(type);
       const value = values[index];
 
-      if (!isDynamicParser(parser, type)) {
+      // If packed mode is enabled, we can skip the dynamic check, as all
+      // values are encoded in the static buffer.
+      if (packed || arrayPacked || !isDynamicParser(parser, type)) {
```

Comment:
Why is the extra `arrayPacked` necessary? 🤔

Reply:
Array values are padded to 32 bytes, but the values are still encoded in place. I needed a way to skip adding the length, but still pad to 32 bytes.

Reply:
Hmm, I wonder if there is a better way 🤔

Reply:
Perhaps open an issue to revisit this?

Reply:
#19